### PR TITLE
Support INVOKE* instructions in CFG

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

File: java-frontend/src/test/java/org/sonar/java/bytecode/cfg/Instructions.java [hunk 1]
```diff
@@ -181,12 +182,12 @@ public Instructions visitMultiANewArrayInsn(String desc, int dims) {
     if (NO_OPERAND_INSN.contains(opcode)) {
       visitInsn(opcode);
     } else {
-      return cfg(opcode, 1, null);
+       return cfg(opcode, 1, null, null);
```
Comment (benzonico) [inline java-frontend/src/test/java/org/sonar/java/bytecode/cfg/Instructions.java]: formatting

File: java-frontend/src/test/java/org/sonar/java/bytecode/cfg/Instructions.java [hunk 2]
```diff
@@ -241,4 +246,36 @@ public Instructions visitMultiANewArrayInsn(String desc, int dims) {
     }
     return cfg();
   }
+
+  static class FieldOrMethod {
```
Comment (benzonico) [inline java-frontend/src/test/java/org/sonar/java/bytecode/cfg/Instructions.java]: Why redefining this class and not use the one defined in CFG Builder ?

File: java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java [hunk 3]
```diff
@@ -311,28 +316,71 @@ public void visitEnd() {
...
+      FieldOrMethod(String owner, String name, String desc, boolean ownerIsInterface) {
+        this.owner = owner;
+        this.name = name;
+        this.desc = desc;
+        this.ownerIsInterface = ownerIsInterface;
+      }
+
+      @Override
+      public boolean equals(Object o) {
+        if (this == o)
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java]: MINOR Missing curly brace. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00121)

File: java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java [hunk 4]
```diff
@@ -311,28 +316,71 @@ public void visitEnd() {
...
+          return true;
+        if (o == null || getClass() != o.getClass())
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java]: MINOR Missing curly brace. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00121)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java (hunk 3 above)
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java]: please fix this one

File: java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java [hunk 5]
```diff
@@ -311,28 +316,73 @@ public void visitEnd() {
...
+      FieldOrMethod(String owner, String name, String desc, boolean ownerIsInterface) {
+        this.owner = owner;
+        this.name = name;
+        this.desc = desc;
+        this.ownerIsInterface = ownerIsInterface;
+      }
+
+      @Override
+      public boolean equals(Object o) {
+        if (this == o) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java]: MINOR Missing curly brace. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00121)

File: java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java [hunk 6]
```diff
@@ -311,28 +316,73 @@ public void visitEnd() {
...
+          return true;
+        }
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java]: MINOR Missing curly brace. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00121)

**Pull request merged**